src/content/developers/tutorials/how-to-write-and-deploy-an-nft.md

```diff
@@ -0,0 +1,334 @@
+
+* Follow [these instructions](https://metamask.zendesk.com/hc/en-us/articles/360015289632-How-to-Export-an-Account-Private-Key) to export your private key from Metamask
+
+* See below to get HTTP Alchemy API URL and copy it to your clipboard
+
+![Copy your Alchemy API URL](https://cdn-images-1.medium.com/max/2000/0*5dqCRVCPa3ca5-aw.gif)
+
+Your .env should now look like this:
+
+    API_URL = "https://eth-ropsten.alchemyapi.io/v2/your-api-key"
+    PRIVATE_KEY = "your-metamask-private-key"
+
+To actually connect these to our code, we’ll reference these variables in our hardhat.config.js file on step 13.
+
+## Step 12: Install Ethers.js
+
+Ethers.js is a library that makes it easier to interact and make requests to Ethreum by wrapping [standard JSON-RPC methods](https://docs.alchemyapi.io/alchemy/documentation/alchemy-api-reference/json-rpc) with more user friendly methods.
+
+Hardhat makes it super easy to integrate [Plugins](https://hardhat.org/plugins/) for additional tooling and extended functionality. We’ll be taking advantage of the [Ethers plugin](https://hardhat.org/plugins/nomiclabs-hardhat-ethers.html) for contract deployment ([Ethers.js](https://github.com/ethers-io/ethers.js/) has some super clean contract deployment methods).
+
+In your project directory type:
+
+    npm install --save-dev @nomiclabs/hardhat-ethers 'ethers@^5.0.0'
+
+We’ll also require ethers in our hardhat.config.js in the next step.
+
+## Step 13: Update hardhat.config.js
+
+We’ve added several dependencies and plugins so far, now we need to update hardhat.config.js so that our project knows about all of them.
+
+Update your hardhat.config.js to look like this:
+
+    /**
+    * @type import('hardhat/config').HardhatUserConfig
+    */
+    require('dotenv').config();
+    require("@nomiclabs/hardhat-ethers");
+    const { API_URL, PRIVATE_KEY } = process.env;
+    module.exports = {
+       solidity: "0.7.3",
+       defaultNetwork: "ropsten",
+       networks: {
+          hardhat: {},
+          ropsten: {
+             url: API_URL,
+             accounts: [`0x${PRIVATE_KEY}`]
+          }
+       },
+    }
+
+## Step 14: Compile our contract
+
+To make sure everything is working so far, let’s compile our contract. The compile task is one of the built-in hardhat tasks.
+
+From the command line run:
+
+    npx hardhat compile
+
+You might get a warning about SPDX license identifier not provided in source file , but no need to worry about that — hopefully everything else looks good! If not, you can always message in the [Alchemy discord](https://discord.gg/u72VCg3).
+
+## Step 15: Write our deploy script
+
+Now that our contract is written and our configuration file is good to go, it’s time to write our contract deploy script.
+
+Navigate to the scripts/ folder and create a new file called deploy.js , adding the following contents to it:
```

Comment:
Navigate to the `scripts/` folder and create a new file called `deploy.js`, adding the following contents to it: